### DISCUSSION TITLE
https on flok-next web server

### DISCUSSION THREAD

**Discussion by aaronsherwood** [munshkr/flok#184] [Q&A] [answered]
Love flok, thank you Damián! How can I enable https on the next-flok web server (and use my own cert and key)?

**Comment by munshkr** [comment]
Thanks! :) I have a work-in-progress branch with optional https server and optional custom SSL certificate and key, will try to get this finished soon.

**Reply by munshkr** [reply]
Working on this here https://github.com/munshkr/flok/issues/188

**Reply by munshkr** [reply, marked as answer]
I just merged #188 on `flok-next` branch. If you're running flok-web directly from the repository, you can now run:

```bash
npm start -- --secure
```
or

```bash
npm start -- --secure --ssl-cert path/to/cert.pem --ssl-key path/to/key.pem
```

If you don't specify a certificate and key files, it will use a self-signed certificate bundled with flok.

If you're using the published `flok-web` package, you'll have to wait a little bit until I publish a new version (probably in a few weeks once I finish a couple of other pending issues).

**Reply by aaronsherwood** [reply]
Awesome! The ssl use works great!

I'm not getting any connectivity with Tidal though. It says it's connected to Superdirt in terminal after pasting and running the repl configuration copied from the webpage, but when i evaluate a line no message comes through on the terminal and no sound happens in supercollider (same thing happens now on [https://next.flok.cc/](https://next.flok.cc/) too). I tried updating the repl by `npm install -g flok-repl` with no improvement (though it now says it doesn't understand the -T command for setting the user's name). I then installed flok-repl@1.0.0-alpha.6, which fixed the -T issue, but still no sound.

Also, FYI, this doesn't work with the latest version of Node. I switched to v18.19.0 and it would work. Here is what I got on node v20:

npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'meyda@5.6.0',
npm WARN EBADENGINE   required: {
npm WARN EBADENGINE     node: '^10 || ^12 || ^13 || ^14 || ^15 || ^16 || ^17 || ^18 || ^19'
npm WARN EBADENGINE   },
npm WARN EBADENGINE   current: { node: 'v20.10.0', npm: '10.2.3' }
npm WARN EBADENGINE }

**Reply by munshkr** [reply]
Hmm yeah I'm not getting any message at the browser either... I'll take a look later today.

About the unsupported engine, it's because of a hydra dependency that is not up-to-date. It's just a warning though, it should still work with Node 20.

**Reply by munshkr** [reply]
Ok, I think it's working now... the problem was that flok-repl didn't want to connect to a server with a self-signed certificate (as it is usually insecure), so I had to force it to ignore that. Can you try pulling again?

I also published a new version of `flok-repl` (1.0.0-alpha.10), it should work with that one (or you can try running it manually from the repo with `bin/flok-repl.js` from the `packages/repl` directory).

**Reply by aaronsherwood** [reply]
Works perfectly, thank you!
